### PR TITLE
feat: Remove request header data when health check is enabled

### LINKED ISSUE #2368: health check has an error message
### Issue description

1、add new route；
2、enable health check and add a header parameter;
3、modify route and remove the header parameter;
4、when i submit , there is an error message displayed in the browser

### Expected behavior

add route success

### How to Reproduce

1、add new route；
2、enable health check；
3、add a header parameter ；
4、submit;
5、modify this route；
6、remove the header parameter；
7、submit;

### Screenshots

image
image
image

### Environment

- apisix version (cmd: `apisix version`):2.11.0
- OS (cmd: `uname -a`):linux
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):openresty/1.19.9.1
- etcd version, if have (cmd: run `etcd --version`):etcd Version: 3.4.13
- apisix-dashboard version, if have:2.11.0
- Browser version, if have: chrome(97.0.4692.71)

### Additional context

I checked the apisix code. req_headers is an array, if req_headers not nil ,then there is at least one value in req_headers.
So I think apisix-dashboard pass apisix an empty array,like this: req_headers[]。
Then apisix validate failed，error message is: Array must have at least 1 items

### PR DESCRIPTION
**Pull request by Si-ege**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #2368
1. Fix an error when the health check request header is empty during upstream configuration .
**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by zaunist** [comment]
Could you fix the failed CI ?

**Comment by Si-ege** [comment]
> Could you fix the failed CI ?

I can try.

File: web/cypress/integration/pluginTemplate/create-plugin-template-with-route.spec.js [hunk 1]
```diff
@@ -129,7 +117,21 @@ context('Create PluginTemplate Binding To Route', () => {
...
+      cy.contains('button', 'Search').click();
+      cy.contains(data.pluginTemplateName)
+        .should('be.visible')
+        .siblings()
+        .contains('Delete')
+        .click();
+      cy.contains('button', 'Confirm').click();
+      cy.get(selector.notification).should('contain', data.pluginTemplateErrorAlert);
+      cy.get(selector.notificationClose).should('be.visible').click();
+    });
```
Comment (Baoyuantop) [inline web/cypress/integration/pluginTemplate/create-plugin-template-with-route.spec.js:134]: Why is this test case included inside other test cases?

**Comment by Baoyuantop** [comment]
OK, we should add a test case for this PR.

1. Go to the Route page and click Create.
2. Fill in the name field and click Next.
3. Fill in the required field and switch Active Health Check.
4. Add Request Headers
5. Submit Route data and should create successfully
6. Configure this route and remove Request Headers in Health Check.
7. The update should work.

**Comment by guoqqqi** [comment]
ping @Si-ege , please update this test case

File: web/src/pages/Upstream/Create.tsx [hunk 2]
```diff
@@ -58,7 +58,11 @@ const Page: React.FC = (props) => {
         });
         return;
       }
-
+      if (data.checks) {
+        if (data.checks.active.req_headers?.length === 0) {
+          delete data.checks.active.req_headers;
```
Comment (guoqqqi) [inline web/src/pages/Upstream/Create.tsx:63]: We prefer not to modify the original data directly

File: web/src/pages/Upstream/Create.tsx [hunk 3]
```diff
@@ -60,9 +61,10 @@ const Page: React.FC = (props) => {
       }
       if (data.checks) {
         if (data.checks.active.req_headers?.length === 0) {
-          delete data.checks.active.req_headers;
+          data.checks.active = omit(data.checks.active, 'req_headers');
         }
       }
+      console.log(data);
```
Comment (guoqqqi) [inline web/src/pages/Upstream/Create.tsx:67]:
```suggestion
```

Comment (guoqqqi) [inline web/src/pages/Upstream/Create.tsx:67]:
```suggestion
```

**Comment by guoqqqi** [comment]
@Si-ege please resolve this conflicts

**Pull request closed without merging**